core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -72,6 +72,7 @@ public class JdbcCatalog extends BaseMetastoreViewCatalog
     implements Configurable<Object>, SupportsNamespaces {
 
   public static final String PROPERTY_PREFIX = "jdbc.";
+  public static final String RETRYABLE_STATUS_CODES = "retryable_status_codes";
```

Comment:
To be consistent, I would move this property in `JdbcUtil` as it's where all properties are located.